lib/engine.js

```diff
@@ -393,77 +397,59 @@ class Engine {
             };
 
          */
-        // if transformer.disableSourceDownload (defaults to false) then set it on input
-        const disableInputDownload = transformer.options && transformer.options.disableSourceDownload;
-        debug('disableInputDownload : ', disableInputDownload);
-        const generateUrl = (input && input.sourceType === TransformerSourceType.URL) 
-                || input.generateUrl; // input.generateUrl will go away
-
-        const downloadAsset = !disableInputDownload && !generateUrl;
-
         // create input object (type Asset)
         let transformerInput = {};
-        // asset needs to be downloaded locally
-        if(downloadAsset && input.url && !input.path) {
-
-            this[INTERNAL].timers.download.start();
-
-            // download input to <baseDirectory>/<stepNumber>-<transformerName>/inputfile and set input.path
-            transformerInput = await getSource(
-                input.url,
-                transformerDirectory.base,
-                false
-            );
-
-            this[INTERNAL].timers.download.stop();
-
-        } // url to asset is available and transformer to use it to process asset (ex: worker-ffmpeg)
-        else if(!downloadAsset && input.url) {
-            // dont download - T.compute will use url to process asset
-            // input.url is already set and is passed on to Asset object 
-            debug(`asset url available and disableInputDownload is ${disableInputDownload} : dont need to do anything`);
-            transformerInput = new Asset(input);
-        } // if (file exists locally && (T.manifest.input.sourceType === 'URL') then upload to tmp cloud
-        else if(!downloadAsset && input.path) {
-            const preSignedUrl = await this.getPresignedUrl(input.path);
-            // TODO: remove this log
-            debug('generated presigned url', preSignedUrl);
-            input.url = preSignedUrl;
-            transformerInput = new Asset(input, input.path);
-
-        } // asset is an intermediate rendition from previous transformer and is already available in input.path
-        else if(input.path) {
-            transformerInput = new Asset(input, input.path);
+        if (input && !input.url && !input.path) {
+            throw new GenericError('No source file accessible.', 'prepareTransformer');
         }
 
-        if (!transformerInput || !transformerInput.path) {
-            throw new GenericError('No source file accessible.');
+        const inputUrl = input.url || '';
+        let protocol = '';
+        if (inputUrl) {
```

Comment:
well lets just say we get an input like this:
```
input: {
                type: 'image/png',
                url: 'invalid url!',
                path: 'valid/path/file.png
                sourceType: 'LOCAL'
            }
```
this will result in an error even though the transformer has a valid path to a local file. its very very edge case, but we may only want to check the validity of the url if we need to